src/caret_analyze/record/period.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2021 Research Institute of Systems Planning, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Optional
+
+from .column import ColumnValue
+from .interface import RecordsInterface
+from .record_factory import RecordFactory, RecordsFactory
+
+
+class Period:
+
+    def __init__(
+        self,
+        records: RecordsInterface,
+        target_column: Optional[str] = None
+    ) -> None:
+        """
+        Constructor.
+
+        Parameters
+        ----------
+        records : RecordsInterface
+            records to calculate period.
+        target_column : Optional[str], optional
+            Column name of timestamps used in the calculation, by default None
+            If None, the first column of records is selected.
+
+        """
+        self._target_column = target_column or records.columns[0]
+        self._target_timestamps: List[int] = []
+        for record in records:
+            if self._target_column in record.columns:
+                timestamp = record.get(self._target_column)
+                self._target_timestamps.append(timestamp)
+
+    def to_records(self) -> RecordsInterface:
+        """
+        Calculate period records.
+
+        Returns
+        -------
+        RecordsInterface
+            period records.
+            Columns
+            - {timestamp_column}
+            - {period_column}
+
+        """
+        records = self._create_empty_records()
+        if not self._target_timestamps:
+            return records
+
+        for i in range(1, len(self._target_timestamps)):
+            records.append(RecordFactory.create_instance(
+                {self._target_column: self._target_timestamps[i],
```

Comment:
I think left column should have former timestamp to make it the same as other APIs (e.g. response time, latency)

```suggestion
                {self._target_column: self._target_timestamps[i-1],
```

Reply:
@hsgwa 
Thank you very much for pointing that out.
I too thought former timestamps should be used, so I modified it: https://github.com/tier4/CARET_analyze/pull/168/commits/2f43a1ddd53a9aa13e188902103db5207e947005

[The previous implementation](https://github.com/tier4/CARET_analyze/blob/3a1ed200e8206fe4ef7eea184325e314b597c493/src/caret_analyze/plot/bokeh/callback_info.py#L132) used a later timestamp, and this modification changes the output results.
So I will notify the user when this Period API is used in the output of the Plot package.